kornia/geometry/robust_estimators/homography.py

```diff
@@ -0,0 +1,93 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from typing import Tuple, Optional
+
+import kornia
+
+TupleTensor = Tuple[torch.Tensor, torch.Tensor]
+
+
+def normalize_points(points: torch.Tensor, eps: float = 1e-6) -> TupleTensor:
+    '''Function, which normalizes points to have zero mean and unit variance
+    and returns corresponding transform
+    See "In Defence of the 8-point Algorithm"
+    http://www.cs.cmu.edu/afs/andrew/scs/cs/15-463/f07/proj_final/www/amichals/fundamental.pdf
+    '''
+    assert len(points.shape) == 3, points.shape
+    assert points.shape[-1] == 2, points.shape
+
+    x_mean = torch.mean(points, dim=-2, keepdim=True)  # Bx1x2
+
+    x_dist = torch.norm(points - x_mean, dim=-1).unsqueeze(-1)  # BxNx1
+    x_dist = torch.mean(x_dist, dim=-2, keepdim=True)  # Bx1x1
+
+    scale = torch.sqrt(torch.tensor(2.)) / (x_dist + eps)  # Bx1x1
+    ones, zeros = torch.ones_like(scale), torch.zeros_like(scale)
+
+    transform = torch.cat([
+        scale, zeros, -scale * x_mean[..., 0],
+        zeros, scale, -scale * x_mean[..., 1],
+        zeros, zeros, ones], dim=-1)  # Bx9
+
+    transform = transform.view(-1, 3, 3)  # Bx3x3
+    transform = transform.detach()
+    points_norm = kornia.transform_points(transform, points)  # BxNx2
+
+    return points_norm, transform
+
+
+def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: torch.Tensor) -> torch.Tensor:
+    '''Function, which finds homography via weighted least squares'''
```

Comment:
add some docs with i/o shape. For functional signature I tend to specify shape in the parameter descrption instead of using `Shape`. Easier to read I guess.